### PR TITLE
Implement class-maps and policy-maps for QOS

### LINKED ISSUE #624: Add support for class-map and policy-map for QOS
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

Implement `class-map` and `policy-map` in AVD to produce following configuration

```eos
! Example 1
class-map type qos match-any TEST
   match vlan 10-100
!
! Example 2
class-map type qos match-any TEST
   match cos 7
!
! Example 3
class-map type qos match-any TEST
   match ip access-group TEST
!
policy-map type quality-of-service PM_TEST
   class TEST
      set dscp af11
   !
   class class-default
!
```

**Describe the solution you'd like**

Add support for the following datamodel in `eos_cli_config_gen`

```yaml
class_maps:
  <class-map name>:
    vlan: <str>
    cos: <str>
    ip:
      access_group: <str>
policy_maps:
  qos:
    <PM name>:
      class:
      set:
        dscp: <str dscp field>
        traffic_class: <0-7>
        drop_precedence: < 1-2>

      # Not Implemented
      police: 
        cir:
        rate:
```

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary

Implement class-maps and policy-maps for QOS

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #624 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Implement templates (eos cli and documentation) for the following datamodel

```yaml
class_maps:
  qos:
    <class-map name>:
      vlan: <str>
      cos: <str>
      ip:
        access_group: <str>
policy_maps:
  qos:
    <PM name>:
      class:
      set:
        dscp: <str dscp field>
        traffic_class: <0-7>
        drop_precedence: < 1-2>
```

## How to test

Look at molecule scenario for

- class_maps
- policy_maps

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/class-map.j2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{# class-map #}
+{% if class_maps is arista.avd.defined %}
+{%     if class_maps.qos is arista.avd.defined %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/class-map.j2]: the natural_sort_filter now also test if the value is defined, so you can remove line 3 `{%     if class_maps.qos is arista.avd.defined %}` is not required!

Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/class-map.j2]: Got it! Fixed

**Comment by ClausHolbechArista** [comment]
@titom73 please review the changes I have done. Only thing missing is readme to reflect policy-map datamodel

**Comment by titom73** [comment]
@ClausHolbechArista : Thanks for the update.

It looks good to me and Readme has been updated accordingly under QOS section.

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**